src/main/groovy/edu/oregonstate/mist/recommendations/resources/RecommendationResource.groovy

```diff
@@ -0,0 +1,116 @@
+package edu.oregonstate.mist.recommendations.resources
+
+import com.google.common.base.Optional
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.recommendations.core.Recommendation
+import edu.oregonstate.mist.recommendations.core.University
+import edu.oregonstate.mist.recommendations.db.RecommendationDAO
+import org.hibernate.validator.constraints.NotEmpty
+
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+
+/**
+ * Recommendation Resource Class
+ */
+@Path('/recommendations/')
+@Produces(MediaType.APPLICATION_JSON)
+class RecommendationResource extends Resource {
+    private final RecommendationDAO recommendationDAO
+
+    public RecommendationResource(RecommendationDAO recommendationDAO) {
+        this.recommendationDAO = recommendationDAO
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<Recommendation> getRecommendations(
+            @QueryParam('by') @NotEmpty String by,
+            @QueryParam('student_type') @NotEmpty String studentType,
+            @QueryParam('province') @NotEmpty String province,
+            @QueryParam('batch') @NotEmpty Integer batch,
+            @QueryParam('lower_limit') @NotEmpty Integer lowerLimit,
+            @QueryParam('upper_limit') @NotEmpty Integer upperLimit,
+            @QueryParam('year') Optional<Integer> year,
+            @QueryParam('major') Optional<String> major,
+            @QueryParam('language') Optional<String> language,
+            @QueryParam('page_size') Optional<Integer> pageSize,
+            @QueryParam('page_number') Optional<Integer> pageNum) {
+
+        List<Recommendation> recommendationList
+        studentType = translateStuType(studentType)
+        province = translateProvince(province)
+        if (by.toLowerCase() == "ranking") {
```

Comment:
Instead of using strings such as: ranking, score-diff, arts and science in this class, store those strings in constants and then use the constants.